autoload/asyncomplete.vim

```diff
@@ -514,9 +514,12 @@ function! asyncomplete#preprocess_complete(ctx, items) abort
         setl completeopt=menuone,noinsert,noselect
     endif
 
-    call asyncomplete#log('core', 'asyncomplete#preprocess_complete calling complete()', a:ctx['startcol'], a:items)
-    call complete(a:ctx['startcol'], a:items)
-endfunction
+    let startcol = a:ctx['startcol']
+    call asyncomplete#log('core', 'asyncomplete#preprocess_complete calling complete()', startcol, a:items)
+    if startcol > 0 " Prevent E578: Not allowed to change text here
```

Comment:
what happens if startcol > total cols should we add the check too?

Reply:
Triggering `complete(100, [...])` on an empty line doesn't cause an error. See my investigation below.